src/web.rs

```diff
@@ -48,3 +51,159 @@ impl WebWindowHandle {
         Self { id }
     }
 }
+
+/// Raw window handle for a Web canvas registered via [`wasm-bindgen`].
+///
+/// ## Construction
+/// ```no_run
+/// # use raw_window_handle::Wbg02CanvasWindowHandle;
+/// # use core::{ffi::c_void, ptr::NonNull};
+/// # fn get_canvas() -> NonNull<c_void> { unimplemented!() }
+/// let obj: NonNull<c_void> = get_canvas();
+/// let mut window_handle = Wbg02CanvasWindowHandle::new(obj);
+/// /* set fields */
+/// ```
+///
+/// [`wasm-bindgen`]: https://crates.io/crates/wasm-bindgen
+#[non_exhaustive]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Wbg02CanvasWindowHandle {
+    /// The object representing the [`HtmlCanvasElement`].
+    ///
+    /// It is implied that this object is registered in the [`wasm-bindgen`] table and is an instance
+    /// of [`HtmlCanvasElement`]. The pointer is a direct reference to a [`JsValue`].
+    ///
+    /// [`HtmlCanvasElement`]: https://docs.rs/web-sys/latest/web_sys/struct.HtmlCanvasElement.html
+    /// [`wasm-bindgen`]: https://crates.io/crates/wasm-bindgen
+    /// [`JsValue`]: https://docs.rs/wasm-bindgen/latest/wasm_bindgen/struct.JsValue.html
+    pub obj: NonNull<c_void>,
+}
+
+impl Wbg02CanvasWindowHandle {
+    /// Create a new handle to an [`HtmlCanvasElement`].
```

Comment:
Nit: Make it clear that we're dealing with a pointer

```suggestion
    /// Create a new handle to a pointer to a [`HtmlCanvasElement`].
```